script/deploy/healthCheck.ts

```diff
@@ -239,278 +239,356 @@ const main = defineCommand({
-        // get address from deploy log
-        const addr = deployedContracts[name]
-        if (!addr) {
-          logError(`Periphery contract ${name} not deployed`)
-          numMissing++
-          continue
-        }
+        consola.info(
+          `Found ${numMissing} missing whitelisted address${
+            numMissing === 1 ? '' : 's'
+          }`
+        )
 
-        // check if address is whitelisted
-        const normalized = getAddress(addr)
-        if (!approvedDexs.includes(normalized)) {
-          logError(`Periphery contract ${name} not approved as a DEX`)
-          numMissing++
-        } else consola.success(`Periphery contract ${name} approved as a DEX`)
-      }
+        //          ╭─────────────────────────────────────────────────────────╮
+        //          │                   Check approved sigs                   │
+        //          ╰─────────────────────────────────────────────────────────╯
 
-      consola.info(
-        `Found ${numMissing} missing dex${numMissing === 1 ? '' : 's'}`
-      )
+        consola.box('Checking DEX signatures approved in diamond...')
+        // Check if function signatures are approved
+        const { sigs } = await import(`../../config/sigs.json`)
 
-      //          ╭─────────────────────────────────────────────────────────╮
-      //          │                   Check approved sigs                   │
-      //          ╰─────────────────────────────────────────────────────────╯
+        // Get all approved signatures from contract
+        const approvedSigs = await getApprovedFunctionSignatures(
+          whitelistManager
+        )
 
-      consola.box('Checking DEX signatures approved in diamond...')
-      // Check if function signatures are approved
-      const { sigs } = await import(`../../config/sigs.json`)
+        // Convert sigs to normalized format for comparison
+        const normalizedConfigSigs = sigs.map((sig) => sig.toLowerCase() as Hex)
+        const normalizedApprovedSigs = approvedSigs.map((sig) =>
+          sig.toLowerCase()
+        )
 
-      // Function to split array into chunks
-      const chunkArray = <T>(array: T[], chunkSize: number): T[][] => {
-        const chunks: T[][] = []
-        for (let i = 0; i < array.length; i += chunkSize)
-          chunks.push(array.slice(i, i + chunkSize))
+        // Find missing sigs in both directions
+        const missingInContract = normalizedConfigSigs.filter(
+          (sig) => !normalizedApprovedSigs.includes(sig)
+        )
+        const extraInContract = normalizedApprovedSigs.filter(
+          (sig) => !normalizedConfigSigs.includes(sig)
+        )
 
-        return chunks
-      }
+        if (missingInContract.length > 0) {
+          logError(
+            `Missing ${missingInContract.length} signatures in contract that are in config:`
+          )
+          missingInContract.forEach((sig) => consola.info(`  ${sig}`))
+        }
+
+        if (extraInContract.length > 0) {
```

Comment:
Are we doing anything with those?
Maybe it would be good to remove the outdated ones automatically to make sure the list always matches?